src/main/python/hydra/lib/hydrabase.py

```diff
@@ -46,14 +46,15 @@ class AppGroup(object):
     group_name:     group name
     analyser:       Analyser class "name" e-g HAnalyser not HAnalyser()
     """
-    def __init__(self, hydra, app_name, group_name, analyser=None):
+    def __init__(self, hydra, app_name, group_name, num_app_instances, analyser=None):
```

Comment:
@sushilks  It is AppGroup class. When we create a group of app instances (using create_app_instances_group()), we pass the number of app instances in the group. In implementation of that function, if you see we create instance of AppGroup. Against every call  of create_app_instances_group(), we create AppGroup instances. It is not ambiguous as we always create a group for particular app (app_name is part of instance). All we are doing here is to keep track of number of app instances in this particular group. Please look at my next comment for its usage.